common/libzkp/impl/src/lib.rs

```diff
@@ -1,76 +1,121 @@
+pub mod proofs;
+pub mod tasks;
+pub mod verifier;
+pub use verifier::{TaskType, VerifierConfig};
 mod utils;
-mod verifier;
 
+use sbv_primitives::B256;
+use scroll_zkvm_types::util::vec_as_base64;
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue;
 use std::path::Path;
+use tasks::chunk_interpreter::{ChunkInterpreter, TryFromWithInterpreter};
 
-use crate::utils::{c_char_to_str, c_char_to_vec};
-use libc::c_char;
-use verifier::{TaskType, VerifierConfig};
-
-/// # Safety
-#[no_mangle]
-pub unsafe extern "C" fn init(config: *const c_char) {
-    let config_str = c_char_to_str(config);
-    let verifier_config = serde_json::from_str::<VerifierConfig>(config_str).unwrap();
-    verifier::init(verifier_config);
-}
-
-/// # Safety
-#[no_mangle]
-pub unsafe extern "C" fn verify_chunk_proof(
-    proof: *const c_char,
-    fork_name: *const c_char,
-) -> c_char {
-    verify_proof(proof, fork_name, TaskType::Chunk)
+/// Turn the coordinator's chunk task into a json string for formal chunk proving
+/// task (with full witnesses)
+pub fn interpret_chunk_task(
```

Comment:
reading.   interpret: Not a name easy to understand.     is it "check_and_fulfill"?